haskell.nix

```diff
@@ -0,0 +1,108 @@
+# A flake-parts module for Haskell cabal projects.
+{ config, lib, flake-parts-lib, ... }:
+
+let
+  inherit (flake-parts-lib)
+    mkSubmoduleOptions
+    mkPerSystemOption;
+  inherit (lib)
+    mkOption
+    mkDefault
+    types;
+in
+{
+  options = {
+    perSystem = mkPerSystemOption
+      ({ config, self', inputs', pkgs, system, ... }: {
+        options.haskellProject = {
+          haskellPackages = mkOption {
+            type = types.anything;
+            description = ''Which Haskell package set to use'';
+            default = pkgs.haskellPackages;
+          };
+          name = mkOption {
+            type = types.str;
+            description = ''Name of the Cabal package'';
+          };
+          root = mkOption {
+            type = types.path;
+            description = ''Path to the Cabal project root'';
+            default = ./.;
```

Comment:
The default value of `./.` may not be correct? Does it refer to where this module lives? Or where the flake.nix that calls this module lives?

Reply:
`./.` is quite low-level. It will give you the location of the `haskell-template` flake.

You can get your caller's `self` from the top-level module arguments (not `perSystem`).

```nix
default = "${self}";
defaultText = "self  # location of your flake";
```